src/index.ts

```diff
@@ -128,38 +135,46 @@ const isHttpsUrl = (maybeURL: string) => {
   }
 };
 
-export function updateElectronApp(opts: IUpdateElectronAppOptions = {}) {
-  // check for bad input early, so it will be logged during development
-  const safeOpts = validateInput(opts);
+export async function updateElectronApp(opts: IUpdateElectronAppOptions = {}): Promise<IUpdater> {
+  return new Promise((resolve, reject) => {
```

Comment:
question: what's the intent around this Promise wrapping? From what I see, we're just manually wrapping the sync code in the Promise in this function, and `initUpdater` is still synchronous.

Reply:
I modified the `initUpdater` to return an updater controller. But when I call `updateElectronApp` and the app is not ready I can't return this controller because `initUpdater` is called inside of a callback. So I used resolve from Promise to avoid this problem and return the updater controller

Reply:
I removed promise wrapping modifying the  behavior of initUpdater (and renamed it to makeUpdater)